yt_dlp/extractor/crunchyroll.py

```diff
@@ -807,30 +828,44 @@ def _real_extract(self, url):
         hardsub_preference = qualities(requested_hardsubs[::-1])
         requested_formats = self._configuration_arg('format') or ['adaptive_hls']
 
-        formats = []
+        available_formats = {}
         for stream_type, streams in get_streams('streams'):
             if stream_type not in requested_formats:
                 continue
             for stream in streams.values():
-                hardsub_lang = stream.get('hardsub_locale') or ''
-                if hardsub_lang.lower() not in requested_hardsubs:
-                    continue
-                format_id = join_nonempty(stream_type, format_field(stream, 'hardsub_locale', 'hardsub-%s'))
                 if not stream.get('url'):
                     continue
-                if stream_type.endswith('hls'):
+                hardsub_lang = stream.get('hardsub_locale') or ''
+                format_id = join_nonempty(stream_type, format_field(stream, 'hardsub_locale', 'hardsub-%s'))
+                available_formats[hardsub_lang] = (stream_type, format_id, hardsub_lang, stream['url'])
+
+        if '' in available_formats and 'all' not in requested_hardsubs:
+            full_format_langs = set(requested_hardsubs)
+            self.to_screen('To get all formats for the hardsub variants use "--extractor-args crunchyrollbeta:hardsub=all"')
```

Comment:
I don't want to make this message too long, but maybe some indication can be made that specific language codes are also valid values for the argument? Or just where to look up the full documentation?

Reply:
Maybe something like `Some alternative formats are simplified. See https://github.com/yt-dlp/yt-dlp#crunchyrollbeta for more info on how to change this behavior.` instead?

Reply:
I'd like the simplest use case to still be spelled out in the message itself...
`To get hardsub formats use "--extractor-args crunchyrollbeta:hardsub=<lang>", where <lang> is a language code or "all". See the README.`?

Reply:
Actually, it might make sense to not show the explanatory message when the user is already supplying the `hardsub` extractor arg? Currently it gets shown everytime.

My current iteration of the message is `To expand certain hardsub formats use "--extractor-args crunchyrollbeta:hardsub=<language_code>". <language_code> can also be "all". See the README for more info.`

Reply:
I think you should show it anyway. People will set it in their config file and then forget about it, and one line of output isn't that obtrustive.

A bit verbose, but otherwise decent.

Reply:
Then I would want to show this message only once, since that should be enough.